http-clients/src/main/java/org/triplea/http/client/web/socket/client/connections/GameToLobbyConnection.java

```diff
@@ -89,4 +91,14 @@ public void close() {
   public void sendChatMessageToLobby(final ChatUploadParams chatUploadParams) {
     lobbyWatcherClient.uploadChatMessage(lobbyClient.getApiKey(), chatUploadParams);
   }
+
+  public void playerJoined(final String gameId, final UserName playerName) {
+    AsyncRunner.runAsync(() -> lobbyWatcherClient.playerJoined(gameId, playerName))
+        .exceptionally(e -> log.log(Level.INFO, "Failed to notify lobby a player connected", e));
```

Comment:
This is something that concerns me a bit. It becomes the client's responsibility to change the online status.
In case the bot loses its internet connection, does the server clear out the entries correctly?
I think it does, but I'd like to be sure I didn't misinterpret something

Reply:
Also I'm curious why you decided to make it the lobbies resposibility to keep track of the users.

Is there anything that speaks against making the lobby "ask" the current users for some bot every time this menu is clicked via websocket or something?

Reply:
`RemoveGame` will remove players from a game: https://github.com/triplea-game/triplea/pull/6488/files#diff-33a1ee87da31c0aeb9cecfd5deeff19cR133

Recall that `RemoveGame` is called on exit of any lobby-connected game. This functionality is not just for bots, but for any lobby-connected game.

Second, if a game dies without invoking `RemoveGame`, we the game listing in `GameListing` will expire from cache. We check for expired cache entries when computing player game participation and will update the server side model when that happens: 
https://github.com/triplea-game/triplea/pull/6488/files#diff-33a1ee87da31c0aeb9cecfd5deeff19cR216

With that said, there is possibility for the server to "miss" "player-join" and "player-left" messages. It'd be nice to have 100% accuracy, but this feature can tolerate some degree of inaccuracy, it does not have to be 100%.

Reply:
> Is there anything that speaks against making the lobby "ask" the current users for some bot every time this menu is clicked via websocket or something?

I suppose ideally we'd have direct communication from the client to the server. Going a websocket route is a bit indirect. The lobby would have to keep track of in-flight requests, ask bot servers to tell them their participants, receive that back, then flush any requests back to a client when the request could be computed. If any server does not respond in a timely way (EG: CPU busy), we could see unexpected issues. We also take an efficiency hit, if every player requested the current games of other players then for each request the lobby would be making a request to every game. With 50 players, making 3 requests to know which games a player is in, that would be 150 requests to each game that is open. The 'which players are in this game' makes sense for it to be direct to a server, but the 'which games is a player is in' is more difficult to answer that way.

Reply:
Thanks for your explanation, makes sense